From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Patric Stout <github@truebrain.nl>
Date: Mon, 29 Jun 2026 20:55:23 +0200
Subject: [PATCH 6/9] chore: fix compiling telemetry module when disabled

---
 include/CcpTelemetry.h | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/include/CcpTelemetry.h b/include/CcpTelemetry.h
index ea0f39d..455bb30 100644
--- a/include/CcpTelemetry.h
+++ b/include/CcpTelemetry.h
@@ -17,7 +17,7 @@
 #ifndef CCP_TELEMETRY_ENABLED
     #if _MSC_VER
 		#define CCP_TELEMETRY_ENABLED 1
-    #else
+     #else
         #define CCP_TELEMETRY_ENABLED 0
 	#endif
 #endif
@@ -39,6 +39,12 @@
 #else
 
 	#define CcpRegisterMutex( m, owner, name )
+	struct TracyCZoneCtx {};
+	namespace tracy {
+		namespace Color {
+			enum { SteelBlue4 = 0 };
+		}
+	}
 #endif // CCP_TELEMETRY_ENABLED
 
 CARBON_CORE_API void CcpRegisterThread( CcpThreadId_t threadId, const char* name );
