diff --git a/base/strings/utf_string_conversion_utils.h b/base/strings/utf_string_conversion_utils.h
index a6ea8d6..7fa2aef 100644
--- a/base/strings/utf_string_conversion_utils.h
+++ b/base/strings/utf_string_conversion_utils.h
@@ -5,6 +5,7 @@
 #ifndef MINI_CHROMIUM_BASE_STRINGS_UTF_STRING_CONVERSION_UTILS_H_
 #define MINI_CHROMIUM_BASE_STRINGS_UTF_STRING_CONVERSION_UTILS_H_
 
+#include <stdint.h>
 #include <string>
 
 namespace base {
diff --git a/base/logging.h b/base/logging.h
index e569539..a42de5d 100644
--- a/base/logging.h
+++ b/base/logging.h
@@ -8,6 +8,7 @@
 #include <assert.h>
 #include <errno.h>
 
+#include <stdint.h>
 #include <limits>
 #include <sstream>
 #include <string>
