From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Patric Stout <github@truebrain.nl>
Date: Sun, 5 Jul 2026 11:45:40 +0200
Subject: [PATCH 09/16] chore: remove dangling >\ at end of comment

---
 trinity/Tr2DynamicRingBuffer.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/trinity/Tr2DynamicRingBuffer.h b/trinity/Tr2DynamicRingBuffer.h
index b04a7253..999aaf56 100644
--- a/trinity/Tr2DynamicRingBuffer.h
+++ b/trinity/Tr2DynamicRingBuffer.h
@@ -10,7 +10,7 @@
 //   dynamic data needs to be passed multiple times per frame for rendering and is
 //   discared right after rendering. With a ring buffer we can avoid excessive discard
 //   locks one would need for use. The basic patter of using this class is:
-//   1. collect data for the buffer>\
+//   1. collect data for the buffer
 //   2. call ringBuffer.PutData to put data into the buffer
 //   3. draw primitives using the ring buffer
 //   4. call ringBuffer.DoneUsingData once the data is no longer needed
